plugins/zstack.py

```diff
@@ -0,0 +1,376 @@
+
+        sacqt = acq.estimateTime(ss)
+        logging.debug("Estimating %g s acquisition for %d streams", sacqt, len(ss))
+
+        dur = sacqt * nsteps + step_time * (nsteps - 1)
+        if last_ss:
+            dur += acq.estimateTime(ss + last_ss) - sacqt
+
+        # Use _set_value as it's read only
+        self.expectedDuration._set_value(math.ceil(dur), force_write=True)
+
+    def _get_live_streams(self, tab_data):
+        """
+        Return all the live streams present in the given tab
+        """
+        ss = list(tab_data.streams.value)
+
+        # On the SPARC, there is a Spot stream, which we don't need for live
+        if hasattr(tab_data, "spotStream"):
+            try:
+                ss.remove(tab_data.spotStream)
+            except ValueError:
+                pass  # spotStream was not there anyway
+
+        for s in ss:
+            if isinstance(s, StaticStream):
+                ss.remove(s)
+        return ss
+
+    def _get_acq_streams(self):
+        """
+        Return the streams that should be used for acquisition
+        return:
+           acq_st (list of streams): the streams to be acquired at every repetition
+           last_st (list of streamsintp): streams to be acquired at the end
+        """
+        if not self._dlg:
+            return []
+
+        live_st = (self._dlg.microscope_view.getStreams() +
+                   self._dlg.hidden_view.getStreams())
+        logging.debug("View has %d streams", len(live_st))
+
+        # On the SPARC, the acquisition streams are not the same as the live
+        # streams. On the SECOM/DELPHI, they are the same (for now)
+        tab_data = self.main_app.main_data.tab.value.tab_data_model
+        if hasattr(tab_data, "acquisitionStreams"):
+            acq_st = tab_data.acquisitionStreams
+            # Discard the acquisition streams which are not visible
+            ss = []
+            for acs in acq_st:
+                if isinstance(acs, stream.MultipleDetectorStream):
+                    if any(subs in live_st for subs in acs.streams):
+                        ss.append(acs)
+                        break
+                elif acs in live_st:
+                    ss.append(acs)
+        else:
+            # No special acquisition streams
+            ss = live_st
+
+        last_ss = []
+        self._acq_streams = acq.foldStreams(ss, self._acq_streams)
+        return self._acq_streams, last_ss
+
+    def start(self):
+        # Fail if the live tab is not selected
+        tab = self.main_app.main_data.tab.value
+        if tab.name not in ("secom_live", "sparc_acqui"):
+            box = wx.MessageDialog(self.main_app.main_frame,
+                       "ZStack acquisition must be done from the acquisition stream.",
+                       "ZStack acquisition not possible", wx.OK | wx.ICON_STOP)
+            box.ShowModal()
+            box.Destroy()
+            return
+
+        # On SPARC, fail if no ROI selected
```

Comment:
Let's also do that on the SECOM confocal FLIM. I think it should be possible to just check if `hasattr(tab.tab_data_model, "roa")` and use that `.roa` VA for the ROI.